theano/gpuarray/opt.py

```diff
@@ -210,6 +211,85 @@ def apply(self, fgraph):
                     0, 'fast_run', 'fast_compile', 'merge')
 
 
+class GraphToGPU(Optimizer):
+    """
+    Transfer the graph as a whole to GPU instead of replacing nodes
+    """
+    def add_requirements(self, fgraph):
+        fgraph.attach_feature(toolbox.ReplaceValidate())
+
+    def apply(self, fgraph):
+        mapping = {}
+        move_to_GPU = True
+
+        # Building a new graph
+        # Iterating through inputs of graph
+        for i in fgraph.inputs:
+            if isinstance(i.type, tensor.TensorType):
+                mapping[i] = GpuFromHost(None)(i)
+            else:
+                mapping[i] = i
+
+        # Iterating through output of all the nodes
+        for n in fgraph.toposort():
+            for o in n.outputs:
+                if isinstance(i.type, tensor.TensorType):
```

Comment:
And yes! I had pasted the output loop inside the node's iteration. Damn.
Now, there aren't any errors, but I get 11 segmentation faults. What could be the reason ?
```
(macvnev) Ramanas-MacBook-Pro:Theano Ramana$ THEANO_FLAGS='mode=FAST_RUN,device=gpu,lib.cnmem=0.5' theano-nose theano/gpuarray/tests/test_opt.py
Using gpu device 0: GeForce GT 750M (CNMeM is enabled with initial size: 50.0% of memory, cuDNN 4007)
Mapped name None to device cuda: GeForce GT 750M
Using cuDNN version 4007 on context None
.........SS......SSSSS..................
----------------------------------------------------------------------
Ran 40 tests in 26.801s

OK (SKIP=7)
Segmentation fault: 11

```